src/Illuminate/Log/LogManager.php

```diff
@@ -0,0 +1,542 @@
+<?php
+
+namespace Illuminate\Log;
+
+use Closure;
+use Throwable;
+use Illuminate\Support\Str;
+use Psr\Log\LoggerInterface;
+use InvalidArgumentException;
+use Monolog\Logger as Monolog;
+use Monolog\Handler\StreamHandler;
+use Monolog\Handler\SyslogHandler;
+use Monolog\Formatter\LineFormatter;
+use Monolog\Handler\ErrorLogHandler;
+use Monolog\Handler\HandlerInterface;
+use Illuminate\Log\Events\MessageLogged;
+use Monolog\Handler\RotatingFileHandler;
+
+class LogManager implements LoggerInterface
+{
+    /**
+     * The application instance.
+     *
+     * @var \Illuminate\Foundation\Application
+     */
+    protected $app;
+
+    /**
+     * The array of resolved logs.
+     *
+     * @var array
+     */
+    protected $logs = [];
+
+    /**
+     * The registered custom driver creators.
+     *
+     * @var array
+     */
+    protected $customCreators = [];
+
+    /**
+     * The Log levels.
+     *
+     * @var array
+     */
+    protected $levels = [
+        'debug' => Monolog::DEBUG,
+        'info' => Monolog::INFO,
+        'notice' => Monolog::NOTICE,
+        'warning' => Monolog::WARNING,
+        'error' => Monolog::ERROR,
+        'critical' => Monolog::CRITICAL,
+        'alert' => Monolog::ALERT,
+        'emergency' => Monolog::EMERGENCY,
+    ];
+
+    /**
+     * Create a new Log manager instance.
+     *
+     * @param  \Illuminate\Foundation\Application  $app
+     * @return void
+     */
+    public function __construct($app)
+    {
+        $this->app = $app;
+    }
+
+    /**
+     * Get a log channel instance.
+     *
+     * @param  string  $driver
+     * @return mixed
+     */
+    public function channel($channel = null)
+    {
+        return $this->driver($channel);
+    }
+
+    /**
+     * Get a log driver instance.
+     *
+     * @param  string  $driver
+     * @return mixed
+     */
+    public function driver($driver = null)
+    {
+        return $this->get($driver ?? $this->getDefaultDriver());
+    }
+
+    /**
+     * Attempt to get the log from the local cache.
+     *
+     * @param  string  $name
+     * @return \Psr\Log\LoggerInterface
+     */
+    protected function get($name)
+    {
+        try {
+            return $this->stores[$name] ?? with($this->resolve($name), function ($monolog) use ($name) {
+                return $this->tap($name, new Logger($monolog, $this->app['events']));
+            });
+        } catch (Throwable $e) {
+            return tap($this->createEmergencyLogger(), function ($logger) use ($e) {
+                $logger->emergency('Unable to create configured logger. Using emergency logger.');
+
+                $logger->emergency($e);
```

Comment:
It would be advisable to pass the exception in the context as monolog provides special exception rendering:
```
$logger->emergency('Unable to create configured logger. Using emergency logger.', ['exception' => $e]);
```

Reply:
Ah OK, Thanks